server/db/Order.js

```diff
@@ -28,16 +29,39 @@ Order.getActiveOrderByUser = (userId) => {
   })
 }
 
-Order.addLineItem = ({userId, product}) => {
-  User.findById(userId, { include: Order })
+Order.createLineItem = ({orderId, productId}) => {
+  LineItem.findOne({
+    where: {
+      productId: productId
+    }
+  })
+    .then(lineItem => {
+      if(lineItem) {
+        lineItem.update({ qty: lineItem.increment() })
+      }
+      else {
+        LineItem.create()
+          .then(lineItem => {
+            Product.findById(productId)
+              .then(product => lineItem.setProduct(product))
+              .then(() => {
+                Order.findById(orderId)
+                  .then(order => lineItem.setOrder(order))
+              })
+          })
+      }
+    })
+}
+
+Order.addLineItem = ({userId, productId}) => {
+  return User.findById(userId, { include: Order })
       .then(user => {
-        Order.getActiveOrderByUser(user.id)
+        Order.getActiveOrderByUser(userId)
```

Comment:
You need to return Order.getActiveOrderByUser(userId)
Otherwise, error doesn't get returned.